### PR TITLE
Document Object Model

### PR DESCRIPTION
**Pull request by Yanyshpolska**
### Document Object Model

[Demo](https://yanyshpolska.github.io/DOM/) |
[Code base](https://github.com/Yanyshpolska/DOM)

The code is submitted in a dedicated feature branch.

Only code files are submitted.

Please, review.

### CODE REVIEW

**Comment by Yanyshpolska** [comment]
Hi. I have changed some variables and function names. 
Please review.

File: submissions/yanyshpolska/dom/js/script.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  pictureOfTea.alt = objectTea.type;
+  pictureOfTea.src = objectTea.image;
+  descriptionOfTea.innerHTML = objectTea.description;
+};
+
+const listOfTeasNames = [];
+
+for (let i = 0; i < document.querySelectorAll("li.menu__item").length; i += 1) {
+  listOfTeasNames[i] = document.querySelectorAll("li.menu__item")[i];
+  listOfTeasNames[i].onclick = doesChangeTea;
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:59]: Read and use [event delegation](https://javascript.info/event-delegation)

File: submissions/yanyshpolska/dom/js/script.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+const descriptionOfTea = document.querySelector("div.more__about");
+
+const doesChangeTea = function (event) {
+  const objectTea = teas.filter(
+    (obj) => obj.type === event.srcElement.innerHTML
+  )[0];
+  titleOfTea.innerHTML = objectTea.type;
+  pictureOfTea.alt = objectTea.type;
+  pictureOfTea.src = objectTea.image;
+  descriptionOfTea.innerHTML = objectTea.description;
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:52]: Don't update elements. Create or append a template with content.

File: submissions/yanyshpolska/dom/js/script.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  },
+];
+
+const titleOfTea = document.querySelector("h2.more__header");
+const pictureOfTea = document.querySelector("img.more__picture");
+const descriptionOfTea = document.querySelector("div.more__about");
+
+const doesChangeTea = function (event) {
+  const objectTea = teas.filter(
+    (obj) => obj.type === event.srcElement.innerHTML
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:47]: I recommend using property `event.target`

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]
Nice work. See some comments below.

**Comment by Yanyshpolska** [comment]
Hi! Thanks for your comments. Reworked as advised. Please review.

File: submissions/yanyshpolska/dom/js/script.js [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+  {
+    type: "Pu-erh",
+    description:
+      "Pu-erh (also spelled pu'er) comes exclusively from China and is famous for its distinctively earthy flavor. Pu-erh is tea that has been fermented, often stored underground for several years. Traditionally, pu-erh is compressed into round cakes and can be very expensive.",
+    image: "./img/puer.jpg",
+  },
+];
+
+const createContent = function (objectTea) {
+  const wholeArticle = document.querySelector("article.more");
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:42]:
```suggestion
  const wholeArticle = document.querySelector(".more");
```

File: submissions/yanyshpolska/dom/js/script.js [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+  wholeArticle.append(descr);
+};
+// ===== initial page
+createContent(teas[0]);
+
+const listOfTeasNames = document.querySelector("ul.menu__items");
+
+listOfTeasNames.onclick = function (event) {
+  const objectTea = teas.filter(
+    (obj) => obj.type === event.target.innerHTML
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:69]: If we have the same name in the result we get more than one type of tea.
Modernize your tea list, and add an `id` or `teaId` property. This field was unique to one type of tea. The id field you can write to a [dataset](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/dataset) of the element. Then you can compare the value from the date and the selected element.

File: submissions/yanyshpolska/dom/js/script.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+  )[0];
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:70]: Avoid construction `[ ]`. If you want find one element try  ti use method find.

File: submissions/yanyshpolska/dom/js/script.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+  const descr = document.createElement("div");
+  descr.classList.add("more__about");
+  descr.innerHTML = `${objectTea.description}`;
+
+  wholeArticle.append(picture);
+  wholeArticle.append(title);
+  wholeArticle.append(descr);
+};
+// ===== initial page
+createContent(teas[0]);
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:63]: Create a variable with an initial value.

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Comment by Yanyshpolska** [comment]
Hi! Thanks for your comments. Reworked as advised. Please review.

File: submissions/yanyshpolska/dom/js/script.js [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+  descr.classList.add("more__about");
+  descr.innerHTML = `${objectTea.description}`;
+
+  wholeArticle.innerHTML = "";
+  wholeArticle.append(picture);
+  wholeArticle.append(title);
+  wholeArticle.append(descr);
+};
+
+const initValue = teas[Math.floor(Math.random() * teas.length)];
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:69]:
```suggestion
const getRandomNumber = () => Math.floor(Math.random() * teas.length);
const initValue = getRandomNumber();
createContent(teas[initValue]);
```

File: submissions/yanyshpolska/dom/js/script.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+createContent(initValue);
+
+const listOfTeasNames = document.querySelector(".menu__items");
+
+listOfTeasNames.onclick = function (event) {
```
Comment (madmaxWMFU) [inline submissions/yanyshpolska/dom/js/script.js:74]: Use `addEventListener`

**Review by madmaxWMFU — CHANGES_REQUESTED** [review]

**Comment by Yanyshpolska** [comment]
Reworked as advised. Please review.

**Review by madmaxWMFU — APPROVED** [review]
Great, I approve of your work. Good luck with future tasks!

**Pull request merged**